api/filters.go

```diff
@@ -414,6 +414,75 @@ func (api *API) addFilterDimension(w http.ResponseWriter, r *http.Request) {
 	api.respond.JSON(ctx, w, http.StatusCreated, resp)
 }
 
+func (api *API) updateFilterDimension(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	filterID := chi.URLParam(r, "id")
+	dimensionName := chi.URLParam(r, "name")
+
+	logData := log.Data{
+		"filter_id":      filterID,
+		"dimension_name": dimensionName,
+	}
+
+	updatedDimension := model.Dimension{
+		Options: []string{},
+	}
+
+	if err := api.ParseRequest(r.Body, &updatedDimension); err != nil {
+		api.respond.Error(
+			ctx,
+			w,
+			http.StatusBadRequest,
+			Error{
+				err:     errors.Wrap(err, "failed to parse update filter request"),
+				logData: logData,
+			},
+		)
+		return
+	}
+
+	// The new dimension won't be present on the dataset (i.e. only `City` will be present, not `Country`),
+	// so instead of validating it against the existing Version, we check to see if the dimension exists in Cantabular.
+	if err := api.validateCantabularDimensionExists(ctx, filterID, updatedDimension.Name); err != nil {
+		api.respond.Error(
+			ctx,
+			w,
+			statusCode(err),
+			Error{
+				err:     errors.Wrapf(err, "error searching for dimension %s in cantabular", updatedDimension.Name),
+				logData: logData,
+			},
+		)
+		return
+	}
+
+	var eTag string
+	if reqETag := api.getETag(r); reqETag != eTagAny {
+		eTag = reqETag
+	}
+
+	newETag, err := api.store.UpdateFilterDimension(ctx, filterID, dimensionName, updatedDimension, eTag)
+	if err != nil {
+		api.respond.Error(
+			ctx,
+			w,
+			statusCode(err),
+			Error{
+				err:     errors.Wrap(err, "unable to update filter dimension"),
+				logData: logData,
+			},
+		)
+		return
+	}
+
+	responseDimension := dimensionItem{}
```

Comment:
Couple of things here.. Can we keep to the `updateFilterDimensionResponse` naming convention here too? And also a smaller thing but does add up over time, it's really in the spirit of Go to keep to shorter variable names as much as possible. Here `resp` would be plenty (and in line with what we call it elsewhere) and earlier you might want to consider if it's necessary to specify `updatedDimension` when maybe `dimension` or `dim` would do.